sphinx_gallery/gen_gallery.py

```diff
@@ -461,41 +472,88 @@ def generate_gallery_rst(app):
 
     for examples_dir, gallery_dir in workdirs:
 
-        examples_dir = os.path.join(app.builder.srcdir, examples_dir)
-        gallery_dir = os.path.join(app.builder.srcdir, gallery_dir)
+        examples_dir_abs_path = os.path.join(app.builder.srcdir, examples_dir)
+        gallery_dir_abs_path = os.path.join(app.builder.srcdir, gallery_dir)
 
-        # Here we don't use an os.walk, but we recurse only twice: flat is
-        # better than nested.
-        this_fhindex, this_costs = generate_dir_rst(
-            examples_dir, gallery_dir, gallery_conf, seen_backrefs)
+        # Create section rst files and fetch content which will
+        # be added to current index file
+        this_fhindex, this_costs, this_toctree, _ = generate_dir_rst(
+            examples_dir_abs_path,
+            gallery_dir_abs_path,
+            gallery_conf,
+            seen_backrefs
+        )
 
         costs += this_costs
-        write_computation_times(gallery_conf, gallery_dir, this_costs)
+        write_computation_times(gallery_conf, gallery_dir_abs_path, this_costs)
 
         # we create an index.rst with all examples
-        index_rst_new = os.path.join(gallery_dir, 'index.rst.new')
+        index_rst_new = os.path.join(gallery_dir_abs_path, 'index.rst.new')
         with codecs.open(index_rst_new, 'w', encoding='utf-8') as fhindex:
             # :orphan: to suppress "not included in TOCTREE" sphinx warnings
-            fhindex.write(":orphan:\n\n" + this_fhindex)
+            fhindex.write(":orphan:\n\n" + this_fhindex + this_toctree)
+
+            # list all paths to subsection index files in this array
+            subsection_index_files = []
 
             for subsection in get_subsections(
-                    app.builder.srcdir, examples_dir, gallery_conf):
-                src_dir = os.path.join(examples_dir, subsection)
-                target_dir = os.path.join(gallery_dir, subsection)
-                this_fhindex, this_costs = \
-                    generate_dir_rst(src_dir, target_dir, gallery_conf,
-                                     seen_backrefs)
-                fhindex.write(this_fhindex)
-                costs += this_costs
-                write_computation_times(gallery_conf, target_dir, this_costs)
+                    app.builder.srcdir, examples_dir_abs_path, gallery_conf):
+                src_dir = os.path.join(examples_dir_abs_path, subsection)
+                target_dir = os.path.join(gallery_dir_abs_path, subsection)
+                subsection_index_files.append(
+                    os.path.join(
+                        '/', gallery_dir, subsection, 'index.rst'
```

Comment:
Should this have been `os.path.sep` rather than `/` (to make things work properly on Windows)? Or something else?

Reply:
You think you are right!

Reply:
Either that or `'/'`.join(..)` would make sense. Do you have a Windows machine or VM where you could test and try to replicate the failure? If not, I can do it

Reply:
Sorry I don't have a Windows machine!
I had a quick look at the rest of the PR to see if other `os.path.sep` were missing, but I guess this would pop out on the build you are currently trying to run.